integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/connector/test/MediationLibraryServiceTestCase.java

```diff
@@ -0,0 +1,378 @@
+    private static final String HELLO_LIB_NAME = "hello";
+
+    private static final String AMAZON_CONNECTOR_ZIP = "amazons3-connector-1.0.4.zip";
+    private static final String AMAZON_CONNECTOR_LIB_QNAME = "{org.wso2.carbon.connector}amazons3";
+    private static final String AMAZON_LIB_NAME = "amazons3";
+
+    private static final String TEMP_CONNECTOR_LIB_QNAME = "{org.wso2.carbon.connector}temp";
+    private static final String TEMP_LIB_NAME = "temp";
+
+    private static final String FAULTY_CONNECTOR_ZIP = "helloworld-connector-1.0.5-SNAPSHOT.zip";
+    private static final String FAULTY_CONNECTOR_LIB_QNAME = "{org.wso2.carbon.connector}helloworld";
+    private static final String FAULTY_LIB_NAME = "helloworld";
+
+    private static final String PACKAGE_NAME = "org.wso2.carbon.connector";
+    private static final String ENABLED = "enabled";
+    private static final String DISABLED = "disabled";
+    private static final String CONNECTOR = "connector";
+    private static final String INVALID_CONNECTOR = "invalid-connector";
+
+    private ServerConfigurationManager serverConfigurationManager;
+    private String resourcePath;
+
+    /**
+     * Deploy and enable the hello connector.
+     *
+     * @throws Exception
+     */
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+
+        super.init();
+        serverConfigurationManager = new ServerConfigurationManager(context);
+        resourcePath = getESBResourceLocation().replace("//", "/") + File.separator + CONNECTOR;
+
+        //upload connector
+        uploadConnector(resourcePath, HELLO_CONNECTOR_ZIP);
+        waitUntilLibAvailable(HELLO_CONNECTOR_ZIP);
+
+        //enable connector
+        updateConnectorStatus(HELLO_CONNECTOR_LIB_QNAME, HELLO_LIB_NAME, PACKAGE_NAME, ENABLED);
+        Thread.sleep(10000);
+
+        serverConfigurationManager.restartGracefully();
+        super.init();
+    }
+
+    /**
+     * Test for connector deploying, enabling and deleting.
+     *
+     * @throws Exception
+     */
+    @Test(groups = "wso2.esb", description = "Test connector deploying, enabling and deleting.")
+    public void deployEnableDeleteConnectorTest() throws Exception {
+
+        String expectedImport =
+                "<import xmlns=\"http://ws.apache.org/ns/synapse\" name=\"amazons3\" package=\"org.wso2.carbon" +
+                        ".connector\" status=\"disabled\"/>";
+
+        //deploy connector
+        uploadConnector(resourcePath, AMAZON_CONNECTOR_ZIP);
+        waitUntilLibAvailable(AMAZON_CONNECTOR_ZIP);
+        Assert.assertFalse(checkAvailabilityInImports(AMAZON_CONNECTOR_LIB_QNAME),
+                           "Connector is already enabled at the deployment.");
+
+        //enable connector
+        updateConnectorStatus(AMAZON_CONNECTOR_LIB_QNAME, AMAZON_LIB_NAME, PACKAGE_NAME, ENABLED);
+        Thread.sleep(5000);
+        Assert.assertTrue(checkAvailabilityInImports(AMAZON_CONNECTOR_LIB_QNAME),
```

Comment:
Remove sleep and wait in a loop until It appear

Reply:
Updated.